static/js/containers/DashboardPage.js

```diff
@@ -229,6 +237,36 @@ class DashboardPage extends React.Component {
     }
   };
 
+  handleCoupon = () => {
+    const { coupons, dispatch, location: { query } } = this.props;
+
+    if (!query.coupon) {
+      // If there's no coupon code in the URL query parameters,
+      // there's nothing to do.
+      return;
+    }
+
+    if ( coupons.fetchPostStatus !== undefined ) {
+      // If we've already launched a coupon fetch, don't launch another one
+      return;
+    }
+
```

Comment:
Yes, that's the race condition. We can't abort fetches, at least not with the fetch API: http://stackoverflow.com/questions/31061838/how-do-i-cancel-an-http-fetch-request